src/operators/Aggregate.cpp

```diff
@@ -0,0 +1,515 @@
+        // all of them at once, since you need to know how many unique values
+        // are in each column to initialize the loop variables.
+        for (auto &col_ref : group_by_refs_) {
+            unique_values_.push_back(get_unique_values(col_ref));
+        }
+
+    }
+
+
+    std::vector<std::shared_ptr<arrow::ArrayBuilder>>
+    Aggregate::get_group_builders() {
+
+        arrow::Status status;
+        std::vector<std::shared_ptr<arrow::ArrayBuilder>> group_builders;
+        for (auto &field : group_type->children()) {
+            switch(field->type()->id()) {
+
+                case arrow::Type::STRING: {
+                    group_builders.push_back(
+                            std::make_shared<arrow::StringBuilder>());
+                    break;
+                }
+                case arrow::Type::INT64: {
+                    group_builders.push_back(
+                            std::make_shared<arrow::Int64Builder>());
+                    break;
+                }
+                default: {
+                    std::cerr << "Aggregate does not support group bys of type "
+                    + field->type()->ToString() << std::endl;
+                }
+            }
+        }
+
+        return group_builders;
+    }
+
+    std::shared_ptr<arrow::ArrayBuilder> Aggregate::get_aggregate_builder
+            (AggregateKernels kernel) {
+
+        arrow::Status status;
+        std::shared_ptr<arrow::ArrayBuilder> aggreagte_builder;
+
+        switch (kernel) {
+            // Returns a Datum of the same type INT64
+            case SUM: {
+                aggreagte_builder = std::make_shared<arrow::Int64Builder>();
+                break;
+            }
+            case MEAN: {
+                aggreagte_builder = std::make_shared<arrow::DoubleBuilder>();
+                break;
+            }
+            case COUNT: {
+                throw std::runtime_error("Count aggregate not supported.");
+            }
+        }
+
+        return aggreagte_builder;
+    }
+
+    std::shared_ptr<arrow::Schema> Aggregate::get_output_schema(
+            AggregateKernels kernel,
+            std::string agg_col_name) {
+
+        arrow::Status status;
+        arrow::SchemaBuilder schema_builder;
+
+        status = schema_builder.AddFields(group_type->children());
+        evaluate_status(status, __FUNCTION__, __LINE__);
+
+        switch (kernel) {
+            // Returns a Datum of the same type INT64
+            case SUM: {
+                status = schema_builder.AddField(
+                        arrow::field(agg_col_name, arrow::int64()));
+                evaluate_status(status, __FUNCTION__, __LINE__);
+                break;
+            }
+            case MEAN: {
+                status = schema_builder.AddField(
+                        arrow::field(agg_col_name, arrow::float64()));
+                evaluate_status(status, __FUNCTION__, __LINE__);
+                break;
+            }
+            case COUNT: {
```

Comment:
Should we add a `default` case switch to catch errors for "unsupported" aggregates?